src/gui/qgsmapcanvas.cpp

```diff
@@ -2109,3 +2122,56 @@ const QgsLabelingEngineSettings &QgsMapCanvas::labelingEngineSettings() const
 {
   return mSettings.labelingEngineSettings();
 }
+
+void QgsMapCanvas::startPreviewJobs()
+{
+  stopPreviewJobs(); //just in case still running
+
+  QgsRectangle mapRect = mSettings.visibleExtent();
+
+  for ( int j = 0; j < 3; ++j )
+  {
+    for ( int i = 0; i < 3; ++i )
+    {
+      if ( i == 1 && j == 1 )
+      {
+        continue;
+      }
+
+
+      //copy settings, only update extent
+      QgsMapSettings jobSettings = mSettings;
+
+      double dx = ( i - 1 ) * mapRect.width();
+      double dy = ( 1 - j ) * mapRect.height();
+      QgsRectangle jobExtent = mapRect;
+      jobExtent.setXMaximum( jobExtent.xMaximum() + dx );
+      jobExtent.setXMinimum( jobExtent.xMinimum() + dx );
+      jobExtent.setYMaximum( jobExtent.yMaximum() + dy );
+      jobExtent.setYMinimum( jobExtent.yMinimum() + dy );
+
+      jobSettings.setExtent( jobExtent );
+      jobSettings.setFlag( QgsMapSettings::DrawLabeling, false );
+
+      QgsMapRendererQImageJob *job = new QgsMapRendererParallelJob( jobSettings );
+      mPreviewJobs.append( job );
+      connect( job, &QgsMapRendererJob::finished, this, &QgsMapCanvas::previewJobFinished );
+      job->start();
+    }
+  }
+}
+
+void QgsMapCanvas::stopPreviewJobs()
+{
+  QList< QgsMapRendererQImageJob * >::const_iterator it = mPreviewJobs.constBegin();
+  for ( ; it != mPreviewJobs.constEnd(); ++it )
+  {
+    if ( *it )
+    {
+      disconnect( *it, &QgsMapRendererJob::finished, this, &QgsMapCanvas::previewJobFinished );
+      connect( *it, &QgsMapRendererQImageJob::finished, *it, &QgsMapRendererQImageJob::deleteLater );
+      ( *it )->cancelWithoutBlocking();
+    }
+  }
+  mPreviewJobs.clear();
```

Comment:
I think the preview jobs are not being deleted - suddenly I am getting big memory leaks when browsing map...